### PR TITLE
improved static library install and packaging

### PR DESCRIPTION
**Pull request by pnav**
Hello,

This pull request improves the build and packaging of Embree as a static library for use by other software (e.g. OSPRay). Currently, even when EMBREE_STATIC_LIB is ON, EMBREE_LIBRARY and EMBREE_LIBRARIES are hard-coded to the system-appropriate shared library name, even when shared libraries are not built.  This pull request corrects this behavior and adds target files for the static library dependencies (put into CMAKE_INSTALL_LIBDIR/cmake/embree-\<version\>) so that all static dependencies can be included, and included in the correct order.

This pull request has been tested successfully against OSPRay 1.7.x build and Galaxy 0.2.x build. The OSPRay build requires a small patch for the static case, I will update this with a link to that pull request once it is submitted.

**Known issues:** CMake `find_package()` reports `sys` and `math` as not found, perhaps due to these names being overloaded in CMake. As a work-around, I add a `link_directories(${EMBREE_ROOT_DIR}/@CMAKE_INSTALL_LIBDIR@)` in the static case in `embree-config.cmake`, which allows `libsys.a` and `libmath.a` to be linked. Since OSPRay checks for the correct include directory, etc, rather than relying on Embree_FOUND, the OSPRay config and build succeeds, and since OSPRay uses `find_package(QUIET)` the user does not see the error message.

In short, this isn't quite as clean as I'd like, but this enables the functionality that is currently broken in Embree.

### CODE REVIEW

**Comment by pnav** [comment]
The OSPRay modification to include static Embree is [pull request 292](https://github.com/ospray/ospray/pull/292).

**Comment by jeffamstutz** [comment]
Hi Paul,

I'm almost finished with a complete "modern CMake" refactoring in OSPRay. With that experience I'll likely (or someone on the team at least) do the same for Embree, where things like ```sys``` and ```math``` could be kept ```PRIVATE``` (and thus not exported) to avoid client projects needing to find them.

Once OSPRay's changes are accepted and in ```devel```, I'll report in on what plans will be for Embree. I'll use this PR as the discussion board wrt revising Embree's CMake, where the plan of attack will guide whether this PR physically gets merged and extended vs. a "from scratch" refactor that solves the issues you mentioned.

Stay tuned!

Cheers,
Jeff
